File: backend/src/main/java/com/example/HopitalPlanningProject/model/Personne.java
package com.example.HopitalPlanningProject.model;

import jakarta.persistence.*;
import lombok.*;

@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@ToString
public class Personne {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "idPersonne")
    private int idPersonne;

    @Column(name = "Nom", nullable = false)
    private String nom;

    @Column(name = "Prénom", nullable = false)
    private String prenom;

    @Column(name = "actif", nullable = false)
    private boolean actif;

    @ManyToOne
    @JoinColumn(name = "idContrat", nullable = false)
    private Contrat contrat;
}

File: frontend/src/models/Personne.js
// models/Personne.js
class Personne {
    constructor(idPersonne, nom, prenom, actif, contrat) {
        this.idPersonne = idPersonne;
        this.nom = nom;
        this.prenom = prenom;
        this.actif = actif;
        this.contrat = contrat;  // Ce serait un objet Contrat
    }
}

export default Personne;
